addons/portal/static/src/js/portal_name_and_signature.js

```diff
@@ -0,0 +1,577 @@
+odoo.define('portal.name_and_signature', function (require) {
+    'use strict';
+
+    var core = require('web.core');
+    var Widget = require('web.Widget');
+
+    var _t = core._t;
+
+    /**
+     * This widget allows the user to input his name and to draw his signature.
+     * Alternatively the signature can also be generated automatically based on
+     * the given name and a selected font, or loaded from an image file.
+     */
+    var NameAndSignature = Widget.extend({
+        template: 'portal.sign_name_and_signature',
+        xmlDependencies: ['/portal/static/src/xml/portal_name_and_signature.xml'],
+        events: {
+            // name
+            'input .o_portal_sign_name_input': '_onInputSignName',
+            // signature
+            'click .o_portal_sign_signature': '_onClickSignature',
+            // draw
+            'click .o_portal_sign_draw_button': '_onClickSignDrawButton',
+            'click .o_portal_sign_draw_clear a': '_onClickSignDrawClear',
+            // auto
+            'click .o_portal_sign_auto_button': '_onClickSignAutoButton',
+            'click .o_portal_sign_auto_select_style a': '_onClickSignAutoSelectStyle',
+            'click .o_portal_sign_auto_font_selection a': '_onClickSignAutoFontSelection',
+            'mouseover .o_portal_sign_auto_font_selection a': '_onMouseOverSignAutoFontSelection',
+            'touchmove .o_portal_sign_auto_font_selection a': '_onTouchStartSignAutoFontSelection',
+            // load
+            'click .o_portal_sign_load_button': '_onClickSignLoadButton',
+            'change .o_portal_sign_load_file input': '_onChangeSignLoadInput',
+        },
+
+        /**
+         * Allows options.
+         *
+         * @constructor
+         * @param {Widget} parent
+         * @param {Object} [options={}]
+         * @param {number} [options.signatureRatio=3.0] - The ratio used when
+         *  (re)computing the size of the signature (width = height * ratio)
+         * @param {string} [options.defaultName=''] - The default name of
+         *  the signer.
+         * @param {string} [options.mode='draw'] - The drawing mode, can be one
+         *  of the following values:
+         *  - 'draw': user draws the signature manually with the mouse
+         *  - 'auto': signature is drawn automatically using the selected font
+         *  - 'load': signature is loaded from an image file
```

Comment:
Instead of duplicating the comments, I would replace the description with @see this.setMode (same for all parameters if possible).

Reply:
Ok. I don't think it's possible for the other parameters, mode is the only one that can be changed in real time and this has its own method.